### DISCUSSION TITLE
Allow steps to be used by multiple workflows

### DISCUSSION THREAD

**Discussion by qwerty287** [woodpecker-ci/woodpecker#2289] [Request a Feature]
### Clear and concise description of the problem

Currently, you have often the same pipeline step in multiple pipelines. For example, Woodpecker itself has three `build-web` steps.

### Suggested solution

Allow to reuse the step and its output. Do not rerun it, but copy the files and start the pipeline then with the copied files. (So that multiple pipelines don't share their files.)

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

**Comment by anbraten** [comment]
linking #1400 #1089
